jetty-deploy/src/main/java/org/eclipse/jetty/deploy/bindings/StandardStarter.java

```diff
@@ -35,12 +36,17 @@
     @Override
     public void processBinding(Node node, App app) throws Exception
     {
+        ContextHandlerCollection contexts = app.getDeploymentManager().getContexts();
+
         ContextHandler handler = app.getContextHandler();
-        
-        // start the handler
-        handler.start();
-        
-        // After starting let the context manage state
-        app.getDeploymentManager().getContexts().manage(handler);
+
+        if (contexts.isRunning() && !handler.isRunning())
+        {
+            // start the handler manually
```

Comment:
Do a `LOG.warn` here that StandardStarter may be removed in a future release.

Reply:
out of scope for this PR.
better as a PR for #2116

Reply:
far enough, so let's just change the test and get this committed.